presto-main/src/test/java/com/facebook/presto/block/TestInt128ArrayBlock.java

```diff
@@ -165,6 +168,36 @@ protected boolean isAlignedLongAccessSupported()
     @Override
     protected boolean isSliceAccessSupported()
     {
-        return false;
+        return true;
+    }
+
+    @Override
+    protected void assertSlicePosition(Block block, int position, Slice expectedSliceValue)
+    {
+        int num128Integers = Math.min(getNum128Integers(expectedSliceValue.length()), block.getPositionCount() - position);
+        for (int offset = 0; offset < num128Integers; offset++) {
+            assertEquals(expectedSliceValue.length(), SIZE_OF_LONG * 2);
+
+            assertEquals(block.getSlice(position, offset, SIZE_OF_LONG * 2), expectedSliceValue.slice(offset, SIZE_OF_LONG * 2));
+            assertEquals(block.getSliceLength(position), SIZE_OF_LONG * 2);
+
+            assertTrue(block.bytesEqual(position, offset, expectedSliceValue, 0, SIZE_OF_LONG * 2));
+            assertFalse(block.bytesEqual(position, offset, Slices.utf8Slice("XXXXXXXXXXXXXXXX"), 0, SIZE_OF_LONG * 2));
+        }
+    }
+
+    @Override
+    protected void assertSlicePositionUnchecked(Block block, int internalPosition, Slice expectedSliceValue)
```

Comment:
@yingsu00 Can this method be implemented as `assertSlicePosition(block, internalPosition - block.getOffsetBase(), expectedSliceValue)`?

Reply:
@mbasmanova Oh yes. I updated the PR with it. Thanks!